super-ddf.c

```diff
@@ -4610,6 +4608,45 @@ static void handle_missing(struct ddf_super *ddf, struct active_array *a, int in
 	}
 }
 
+const char *
+array_state_str(enum array_state s)
+{
+	const char *str = NULL;
+	switch (s) {
+	case clear:
+		str = "clear";
+		break;
+	case inactive:
+		str = "inactive";
+		break;
+	case suspended:
+		str = "suspended";
+		break;
+	case readonly:
+		str = "readonly";
+		break;
+	case read_auto:
+		str = "read_auto";
+		break;
+	case clean:
+		str = "clean";
+		break;
+	case active:
+		str = "active";
+		break;
+	case write_pending:
+		str = "write_pending";
+		break;
+	case active_idle:
+		str = "active_idle";
+		break;
+	case bad_word:
+		str = "bad_word";
+		break;
+	}
+	return str;
```

Comment:
Here, if it falls through the switch statement, do you want to return "*INVALID*" or something instead of NULL?